code/modules/mob/living/carbon/human/whisper.dm

```diff
@@ -109,24 +109,18 @@
 			message = replacetext(message, "u", "�")
 			message = replacetext(message, "b", "�")
 
-	var/list/listening = hearers(message_range, src)
+	var/list/listening = get_mobs_or_objects_in_view(message_range, src)
 	listening |= src
 
 	//ghosts
-	for (var/mob/M in dead_mob_list)	//does this include players who joined as observers as well?
+	for (var/mob/M in dead_mob_list)
 		if (!(M.client))
 			continue
 		if(M.stat == DEAD && M.is_preference_enabled(/datum/client_preference/ghost_ears))
 			listening |= M
 
-	//Pass whispers on to anything inside the immediate listeners.
-	for(var/mob/L in listening)
-		for(var/mob/C in L.contents)
-			if(istype(C,/mob/living))
-				listening += C
-
 	//pass on the message to objects that can hear us.
-	for (var/obj/O in view(message_range, src))

```

Comment:
Another unnecessary call to view, as we've already gotten a list of objects in listeners, as GMOOIV returns objects and mobs both.